source/_docs/guides/jenkins.md

```diff
@@ -0,0 +1,310 @@
+
+
+7. In the **Properties File Path** field, enter the path to your Jenkins environment variables file. On Debian-based systems, it's usually `/var/lib/jenkins/envVars.properties`. If it doesn't already exist, create an empty file.
+
+
+8. In the **Properties Content** field, add the following variable one per line, with no quotation marks:
+
+    - **SITE_ID=**your-site-name
+
+    ![Env vars view](/source/docs/assets/images/integrations/env_vars.png)
+
+9. Check "Use secret text(s) or file(s)" option. A new button called **Bindings** will appear. add a secret text binding. Name it **TERMINUS_TOKEN** and select the Terminus secret text credential from the dropdown:
+
+    ![Bindings](/source/docs/assets/images/integrations/jenkins-binding.png)
+
+### Add Build Steps
+
+Under the **Build** tab is a button labeled **Add build step**. These tasks will execute in sequence, and the job will quit if any fail. Add these in separate build steps, selecting **Execute shell** for all but steps 1 and 6.
+
+1. Set build status to "pending" on GitHub commit
+
+2. Jenkins logs into Pantheon:
+
+    ```nohighlight
+    #!/bin/bash
+    echo "Logging into Terminus"
+    terminus auth:login --machine-token=${TERMINUS_TOKEN}
+    ```
+
+3. Verifies the dev site is awake and in git mode. Note that these are separate build steps:
+
+    ```nohighlight
+    echo "Waking Dev environment."
+    terminus env:wake -n ${SITE_ID}.dev
+    ```
+
+    ```nohighlight
+    echo "Setting site to git mode."
+    terminus connection:set ${SITE_ID}.dev git
+    ```
+
+4. Jenkins creates a Multidev and pushes the new code to this environment
+
+    ```nohighlight
+    echo "Creating multidev"
+    cd ${WORKSPACE}
+    terminus build:env:create ${SITE_ID}.dev ci-${BUILD_ID} --yes
+    ```
+
+    ```nohighlight
+    echo "Run database updates and clear cache"
+    terminus drush -n ${SITE_ID}.ci-${BUILD_ID} -- updatedb -y
+    terminus drush ${SITE_ID}.ci-${BUILD_ID} cr
+    ```
+
+5. Then the test suite we include with the example is run.
+
+    ```nohighlight
+    echo "Running behat"
+    TERMINUS_ENV=ci-$BUILD_ID TERMINUS_SITE=$SITE_ID $WORKSPACE/tests/scripts/run-behat
+    ```
+
+6. Add one **Conditional step (single)** build task, to merge the code from the Pantheon Multidev to the pantheon/master, i.e. your dev site on Pantheon. This will only happen when changing the master branch.
+
+    - Under **Run?** select **Regular expression match**.
+
+        - **Expression**: `(?i).*origin/master.*`
+
+        - **Label**: `${ENV,var="GIT_BRANCH"}`
+
+    - Under **Builder** select **Execute Shell**
+
+        - Command:
+
+                echo "Merging multi-dev changes to master"
+                terminus build:env:merge -n ${SITE_ID}.ci-${BUILD_ID} --yes
+
+    Your conditional step should look like this:
+    ![Conditional Step](/source/docs/assets/images/integrations/jenkins-conditional.png)
+
+7. Finally, a cleanup task:
```

Comment:
Suggest explaining what this clean up task will do to both the Pantheon site and the GitHub repo